src/hit/api/evaluator/homomorphic.cpp

```diff
@@ -238,6 +259,19 @@ namespace hit {
         return backend_encryptor->object;
     }
 
+    Bootstrapper &HomomorphicEval::get_bootstrapper() {
+        if (!(context->btp_params.has_value())) {
+            LOG_AND_THROW_STREAM("CKKS parameters do not specify bootstrapping parameters.");
+        }
+
+        if (backend_bootstrapper.get() == nullptr || backend_bootstrapper->params != context->params) {
```

Comment:
This is trippy, but awesome. This uses a C++ smart pointer called `thread_specific_ptr`, which does basically what it sounds like. Here, we create a new object if either:
- the pointer for this thread has not been initialized OR
- the parameters for this thread are different from the parameters of the object currently in the smart pointer

The latter condition is necessary because C++ manages the threads. In the tests, we use this evaluator with many different parameters, but C++ was keeping the thread alive and therefore trying to reuse the object. But the object is parameter-specific, so we have to make sure that the objects parameters match the evaluator's parameters.

Reply:
This is already explained in even greater detail in a comment a few lines above this block: https://github.com/awslabs/homomorphic-implementors-toolkit/blob/wip-lattigo-backend-bootstrapping/src/hit/api/evaluator/homomorphic.cpp#L217

Reply:
Opened an issue to track this: #146